### ISSUE TITLE
Deletion timer for messages per chat

### ISSUE THREAD

**Issue by blauertee** [iNPUTmice/Conversations#3880]
**Advancement Proposal:** On device deletion timer per chat. Like this (Example from Signal):

![image](https://raw.githubusercontent.com/blauertee/images/master/signal_screenshot_message_deletion_timer_1.png)

**Why I think this feature is important:**

Jabber/XMPP should be the way to go for activists around the globe who are fighting against suppressive governments because:
- OMEMO end to end encryption
- no fear of a big Platform cooperating with the state (WhatsApp, Signal, Telegram etc ...)
- No phone numbers to be identified by

But without this feature and just one person not clearing their chat history in time, if just one device gets into the wrong hands all is lost.

I hope you can follow my reasoning and thank you for developing this beautiful app.

**Comment by licaon-kter** [comment]
https://github.com/iNPUTmice/Conversations/issues/2514

**Comment by blauertee** [comment]
This is just sad. I was about to recommend conversations to a group around 200 activists (I do process impovement work for them, which includes thinking about how to communicate) just this little tile is missing. And all of thing Big-Platform-Privicy oriented Messengers do have it.

It makes me almost think about forging out of spite ... But surely I can't do this alone :D

**Comment by blaueente** [comment]
Such a feature would even provide a huge benefit if it is not enforced, see my other comment: https://github.com/iNPUTmice/Conversations/issues/2371#issuecomment-716062864_

**Comment by blauertee** [comment]
I'll try to be more specific, because I completely understand that enforcing the feature on both sides is not possible via XMPP/Jabber and to hack it into or on top of the protocol would cause a huge amount of problems, but this is not what I want and this feature is still pressing to me.

Use case:
- Go tab onto the three dot menu of any chat
- Select deletion timer for local chat History
- Popup appears where you can choose an amount of time x
- (Optional: Little checkbox "Notify conversation participants that I activated deletion timer)
- Tap OK
- (Optional: Automated message is send saying "\<username\> activated local deletion timer via <a href="https://conversations.im">conversations.im</a>. If this chat requires a lot of privacy you maybe should do the same."

Done.


And from now on conversations will delete messages after x amount of time. Since every message already has a timestamp this should be easy. If you don't want message that where sent before the deletion timer was set to not be affected you can just save a timestamp in the same place where you keep track of weather the timer is activated at all. It's also efficient because you always only have to check the timestamp of the first message that was sent after the deletion timer was set. 
When you turn of the deletion timer nothing will be deleted anymore the user can manually take care of this I'd say.

**Comment by iNPUTmice** [comment]
@blauertee you can already delete local history after a certain time out. Settings / Expert Settings / Security / Automatic message deletion.

**Comment by blauertee** [comment]
@iNPUTmice Yes, this is for every chat. But I have some chat's that require more privacy than others. Some where I really want to keep the history of the conversation and some where I definitely don't want to have it on my device.

**Comment by blauertee** [comment]
@iNPUTmice Do you think this feature would be useful at all? Is it just a matter of labour capacity / priorities to not implement it. Or are you convinced nobody needs it and you don't what to have it in the app?

If it's not the latter I might be interested in helping to implement it. Of course I'd have look into the app more to see weather my capabilities and time capacities are sufficient. But I'd do that and then come back to you if this is wanted.
